kotlin-analysis-statistic/src/main/kotlin/org/jetbrains/research/ml/kotlinAnalysis/BinaryExpressionRangesFqPsiAnalyzer.kt

```diff
@@ -0,0 +1,30 @@
+package org.jetbrains.research.ml.kotlinAnalysis
+
+import org.jetbrains.kotlin.idea.caches.resolve.resolveToCall
+import org.jetbrains.kotlin.psi.KtBinaryExpression
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameOrNull
+
+/**
+ * Analyzer for binary expressions.
+ * Determines whether an expression is a range. If it is, determines the type of the range.
+ * This class uses fq names for analysis, so it works correctly only on projects with dependencies.
+ */
+object BinaryExpressionRangesFqPsiAnalyzer : PsiAnalyzer<KtBinaryExpression, RangeType> {
+    override fun analyze(psiElement: KtBinaryExpression): RangeType {
+        val expressionFqName = psiElement.resolveToCall()?.resultingDescriptor?.fqNameOrNull().toString()
```

Comment:
Maybe is it better to throw error() if we can not resolve this call? it means we did not open the project correctly